### PR TITLE
Automatic documentation generation

### PR DESCRIPTION
**Pull request by lightsprint09**
Fixes issues: #51 

New feature:
Generate documentation & push them to gh-pages

#### Make sure to check all boxes before merging

- [x] Method/Class documentation
- [x] README.md documentation
- [x] Unit tests for new features/regressions

### CODE REVIEW

File: Source/NetworkService.swift [hunk 1]
```diff
@@ -46,6 +48,27 @@ public final class NetworkService: NetworkServiceProviding {
         self.networkResponseProcessor = NetworkResponseProcessor()
     }
     
+    /**
+     Fetches a resource asynchronously from remote location. Completion and Error block will be called on the main thread.
+     
+     ```swift
+     
+     let networkService: NetworkServiceProviding = //
+     let resource: Ressource<String> = //
```
Comment (ahersch) [inline Source/NetworkService.swift]: There's a typo: Resource<String> (only with one s).
This applies also for all other documentation where this is mentioned.

File: Source/NetworkServiceProviding.swift [hunk 2]
```diff
@@ -29,12 +29,49 @@ import Foundation
...
 
 /**
- `NetworkServiceProviding` provides access to remote resources.
+ `NetworkServiceProviding` provides access to remote resources. It should be used as an abstraction for all outgoing network resource.
+ 
+ 
+ **Abstract different layers**
+ 
+Builing layerd network abstraction can be very easy, when all layers conform to `NetworkServiceProviding`.
+ By doing so composing different features inot a layerd system becomes very easy.
```
Comment (ahersch) [inline Source/NetworkServiceProviding.swift]: Typo: into

File: Source/Resource.swift [hunk 3]
```diff
@@ -25,16 +25,25 @@
...
+    /**
+     The request to get the remote data payload
+     */
     public let request: URLRequestConvertible
+    /**
+     Parses data into given Model
+     */
     public let parse: (_ data: Data) throws -> Model
     
+    /// Creates a type safe resource, which can be used to fetch it with `NetworkServiceProviding`
```
Comment (ahersch) [inline Source/Resource.swift]: Would go for the common notation with /** ... */ here

File: Source/RetryNetworkService.swift [hunk 4]
```diff
@@ -60,6 +60,27 @@ public final class RetryNetworkService: NetworkServiceProviding {
...
+     
+     networkService.request(resource, onCompletion: { htmlText in
+     print(htmlText)
+     }, onError: { error in
+     //Handle errors
+     })
+     ```
+     
+     - parameter resource: The resource you want to fetch.
+     - parameter onComplition: Callback which gets called when fetching and tranforming into model succeeds.
```
Comment (ahersch) [inline Source/RetryNetworkService.swift]: Typo: onCompletion

File: Source/URLRequestConvertible.swift [hunk 5]
```diff
@@ -28,14 +28,18 @@
 
 import Foundation
 
+/// An abstraction for types which can be represented as an URLRequest
```
Comment (ahersch) [inline Source/URLRequestConvertible.swift]: Also use /** ... */ notation here

File: Source/Resource.swift (hunk 3 above)
Comment (lightsprint09) [reply Source/Resource.swift]: This is the syntax which is autogenerated by Xcode

**Pull request merged**